### PR TITLE
Restart php-fpm if present

### PR DESCRIPTION
**Pull request by t2d**
- Detect services automatically
- Restart apache2 and php-fpm if they are present

### CODE REVIEW

File: handlers/main.yml [hunk 1]
```diff
@@ -17,4 +17,12 @@
   ansible.builtin.service:
     name: apache2
     state: restarted
-  when: roundcube_apache | bool
+  when: ansible_facts.services is search("apache2.service")
+  notify:
+    - Restart php-fpm
```
Comment (doobry-systemli) [inline handlers/main.yml:22]: Technically we don't have to restart apache if `php-fpm` is used. Also, you might run `php-fpm` with another reverse proxy. So I'd suggest to just always notify `Restart apache`  and `Restart php-fpm` when necessary.

**Review by doobry-systemli — CHANGES_REQUESTED** [review]
Nice, thanks for tackling this. Also great to get rid of `roundcube_apache`. I have one comment though.

File: handlers/main.yml (hunk 1 above)
Comment (t2d) [reply handlers/main.yml]: Thank you for your feedback @doobry-systemli . What do you think of this more generic solution? Apache2 might be restarted without a reason, but I don't see this as a big problem?

**Review by doobry-systemli — APPROVED** [review]
I like :heart: I didn't know about the `listen` option in handlers so far, but it's a nice and clean solution.

**Pull request merged**